packer/build_ova.sh

```diff
@@ -0,0 +1,343 @@
+                                       if true, the OVA can be specified IP during deployment. only supported by vCenter"
+}
+
+###################################################
+
+INFO_HEADER="[Info]"
+WARNING_HEADER="[Warning]"
+ERROR_HEADER="[Error]"
+
+# early exit on command failure
+set -e
+
+varDefine(){
+    VM_NAME=rackhd-${OS_VER}
+    BASENAME=${VM_NAME}
+    OVA="${BASENAME}.ova"
+    # default is output-${type}. you can customized in packer's json by "output_directory" param
+    VMDIR=output-vmware-vmx # output of Build VMWare from VMX
+}
+
+prepareMaterials(){
+    set +e
+    pkill packer
+    pkill vmware
+    set -e
+    # Prepare packer and cfg file
+    CFG_FILE=template.cfg
+    # parameter file pass to packer
+    echo {                                                     > $CFG_FILE
+    echo  \"playbook\": \"${ANSIBLE_PLAYBOOK}\",              >> $CFG_FILE
+    echo  \"vm_name\": \"${VM_NAME}\",                        >> $CFG_FILE
+    if [  -n "${DEBIAN_REPOSITORY}" ];  then
+         echo  \"deb_repository\": \"${DEBIAN_REPOSITORY}\",  >> $CFG_FILE
+    fi
+    echo  \"rackhd_version\": \"${RACKHD_VERSION}\"           >> $CFG_FILE
+    echo }                                                    >> $CFG_FILE
+
+    PACKER=packer
+    if [ -x /opt/packer/packer ]
+    then
+        PACKER=/opt/packer/packer
+    fi
+
+    # Enable Verbose Packer Logging
+    # see https://www.packer.io/docs/other/environmental-variables.html for details
+    export PACKER_LOG=1
+    export PACKER_LOG_PATH=./packer-debug.log
+    export PACKER_NO_COLOR=1  # for Jenkins usage. if manual run, suggest to turn color on (set to 0)
+    export PACKER_CACHE_DIR=$HOME/.packer_cache
+
+    # Check Free Disk Space,  VMWare Workstation may stuck if disk space too small
+    fd_in_kb=$(df  .  | awk '/^\/dev/ {print $4}')
+    fd_thres=$(expr 1024 '*' 1024 '*' 8)  # set to 8G as threshold.
+    if [ $fd_in_kb -lt  $fd_thres ]
+    then
+        echo "The Free Up Disk Space($fd_in_kb KB) is not suffcient(recommended to $fd_thres KB). it may cause VMWare Workstation to stuck."
+        exit 2
+    fi
+}
+
+packerBuildOVA(){
+    #packer template file
+    PACKER_TEMP=template-${OS_VER}.json
+
+    # execute 'packer build'
+    if [ "$BUILD_STAGE" == "BUILD_TEMPLATE" ]; then
+        #Build from  iso, create a Pre-build Cache ( vmware:iso-->vmx, virtualbox: iso-->ovf )
+        $PACKER build --force --only=vmware-iso  --var-file=$CFG_FILE  ${PACKER_TEMP}  | tee packer-install.log
+    else
+        if [ "$BUILD_STAGE" == "BUILD_FINAL" ]; then
+             #Build from cache template, and output the final image .(vmware: vmx-->ova, virtualbox: ovf-->box)
+             mkdir output-vmware-iso && mv $CACHE_IMAGE_DIR/* ./output-vmware-iso
+             $PACKER build --force --only=vmware-vmx  --var-file=$CFG_FILE  --var "playbook=${ANSIBLE_PLAYBOOK}_mini" ${PACKER_TEMP}   | tee packer-install.log
```

Comment:
that was my bad , to add tricky ```_mini``` before..
so can we directly use the passing in ANSIBLE_PLAYBOOK without add a surfix  ?
and don't forget to change where build_ova.sh is invoked from ```rackhd_package``` to ```rackhd_package_mini```